### PR TITLE
PoC: using local abstractions sources instead of packages

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
[Abstractions lib sources](https://github.com/opensearch-project/opensearch-net-abstractions)https://github.com/[opensearch-project/opensearch-net-abstractions](https://github.com/opensearch-project/opensearch-net-abstractions) were copied into .net client to use in tests. No need any more to store local nuget packages of abstractions, release them or worry about outdated/broken references to them.
Few fixes were made to let workflows pass.

In case if this approach approved this PR should be merged to upstream/main after closing #51.

### CODE REVIEW

File: abstractions/Directory.Build.props [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    <MinVerMinimumMajorMinor>0.1</MinVerMinimumMajorMinor>
+
+    <LangVersion>latest</LangVersion>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageReference Include="MinVer" Version="2.3.1" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies.net461" Version="1.0.0" PrivateAssets="all"/>
+  </ItemGroup>
+</Project>
```
Comment (MaxKsyunz) [inline abstractions/Directory.Build.props]: newline would be nice.

File: abstractions/dotnet-tools.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      ]
+    },
+    "nupkg-validator": {
+      "version": "0.4.0",
+      "commands": [
+        "nupkg-validator"
+      ]
+    }
+  }
+}
```
Comment (MaxKsyunz) [inline abstractions/dotnet-tools.json]: newline would be nice.

File: abstractions/global.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "sdk": {
+    "version": "5.0.100",
+    "rollForward": "latestFeature",
+    "allowPrerelease": false
+  }
+}
```
Comment (MaxKsyunz) [inline abstractions/global.json]: newline would be nice.

File: abstractions/src/Directory.Build.props [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    <Copyright>OpenSearch</Copyright>
+    <PackageLicenseExpression>Apache-2.0</PackageLicenseExpression>
+    <RepositoryUrl>https://github.com/opensearch-project/opensearch-net-abstractions</RepositoryUrl>
+    <RepositoryType>Git</RepositoryType>
+    <PackageProjectUrl>https://github.com/opensearch-project/opensearch-net-abstractions</PackageProjectUrl>
+    <PackageReleaseNotes>https://github.com/opensearch-project/opensearch-net-abstractions/releases</PackageReleaseNotes>
+    <PackageIconUrl>https://raw.githubusercontent.com/opensearch-project/opensearch-net/main/build/nuget-icon.png</PackageIconUrl>
+
+    <SignAssembly>true</SignAssembly>
+    <AssemblyOriginatorKeyFile>..\..\build\keys\keypair.snk</AssemblyOriginatorKeyFile>
```
Comment (MaxKsyunz) [inline abstractions/src/Directory.Build.props]: Could you make this absolute path?

It's a good convention and is already used in this file.

**Comment by Yury-Fridlyand** [comment]
**UPDATED**
Output from `git diff --diff-filter=M opensearch-net-abstractions/ opensearch-net/abstractions/`
```patch
diff --git a/opensearch-net-abstractions/README.md b/opensearch-net/abstractions/README.md
index 2765381..88b8405 100755
--- a/opensearch-net-abstractions/README.md
+++ b/opensearch-net/abstractions/README.md
@@ -17,8 +17,6 @@

 You've reached the home repository for several auxiliary projects from the .NET team within OpenSearch.

-Pre-release packages of these projects are available at [OpenSearch NuGet Package Repository](https://www.github.com/opensearch-project/packages). See [GitHub documentation](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-nuget-registry#installing-a-package) for instructions on how to access them.
-
 Current projects:

 ### [OpenSearch.OpenSearch.Managed](src/OpenSearch.OpenSearch.Managed/README.md)
diff --git a/opensearch-net-abstractions/src/Directory.Build.props b/opensearch-net/abstractions/src/Directory.Build.props
index 968df46..9612245 100755
--- a/opensearch-net-abstractions/src/Directory.Build.props
+++ b/opensearch-net/abstractions/src/Directory.Build.props
@@ -5,10 +5,7 @@
     <Authors>OpenSearch Project and contributors</Authors>
     <Copyright>OpenSearch</Copyright>
     <PackageLicenseExpression>Apache-2.0</PackageLicenseExpression>
-    <RepositoryUrl>https://github.com/opensearch-project/opensearch-net-abstractions</RepositoryUrl>
     <RepositoryType>Git</RepositoryType>
-    <PackageProjectUrl>https://github.com/opensearch-project/opensearch-net-abstractions</PackageProjectUrl>
-    <PackageReleaseNotes>https://github.com/opensearch-project/opensearch-net-abstractions/releases</PackageReleaseNotes>


     <SignAssembly>true</SignAssembly>
@@ -19,19 +16,6 @@
     <PackageIcon>nuget-icon.png</PackageIcon>

   </PropertyGroup>
-
-  <ItemGroup>
-    <Content Include="../../nuget-icon.png" CopyToOutputDirectory="PreserveNewest">
-      <Link>nuget-icon.png</Link>
-      <Pack>True</Pack>
-      <PackagePath>nuget-icon.png</PackagePath>
-    </Content>
-    <None Include="../../LICENSE.txt" CopyToOutputDirectory="PreserveNewest">
-      <Pack>True</Pack>
-      <PackagePath>LICENSE.txt</PackagePath>
-    </None>
-  </ItemGroup>
-
   <ItemGroup>
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All"/>
     <PackageReference Include="ConfigureAwaitChecker.Analyzer" Version="4.0.0" PrivateAssets="All" />
diff --git a/opensearch-net-abstractions/src/OpenSearch.OpenSearch.Ephemeral/OpenSearch.OpenSearch.Ephemeral.csproj b/opensearch-net/abstractions/src/OpenSearch.OpenSearch.Ephemeral/OpenSearch.OpenSearch.Ephemeral.csproj
old mode 100644
new mode 100755
index a4ec93b..994e437
--- a/opensearch-net-abstractions/src/OpenSearch.OpenSearch.Ephemeral/OpenSearch.OpenSearch.Ephemeral.csproj
+++ b/opensearch-net/abstractions/src/OpenSearch.OpenSearch.Ephemeral/OpenSearch.OpenSearch.Ephemeral.csproj
@@ -4,6 +4,7 @@
     <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
     <Description>Provides an EphemeralCluster implementation that can download/bootstrap/run a throwaway customizable OpenSearch cluster</Description>
     <PackageTags>opensearch,opensearch,cluster,ephemeral</PackageTags>
+    <IsPackable>false</IsPackable>
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="SharpZipLib.NETStandard" Version="1.0.7" />
@@ -12,4 +13,4 @@
   <ItemGroup>
     <ProjectReference Include="..\OpenSearch.OpenSearch.Managed\OpenSearch.OpenSearch.Managed.csproj" />
   </ItemGroup>
-</Project>
\ No newline at end of file
+</Project>
diff --git a/opensearch-net-abstractions/src/OpenSearch.OpenSearch.Managed/OpenSearch.OpenSearch.Managed.csproj b/opensearch-net/abstractions/src/OpenSearch.OpenSearch.Managed/OpenSearch.OpenSearch.Managed.csproj
old mode 100644
new mode 100755
index 1778c51..91a10c3
--- a/opensearch-net-abstractions/src/OpenSearch.OpenSearch.Managed/OpenSearch.OpenSearch.Managed.csproj
+++ b/opensearch-net/abstractions/src/OpenSearch.OpenSearch.Managed/OpenSearch.OpenSearch.Managed.csproj
@@ -7,13 +7,13 @@
       Also ships with an cluster abstraction that can start one or more OpenSearchNode's
     </Description>
     <PackageTags>opensearch,opensearch,cluster,observable,rx</PackageTags>
+    <IsPackable>false</IsPackable>
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="Proc" Version="0.6.1" />
-    <!--<ProjectReference Include="..\..\..\..\proc\src\Proc\Proc.csproj" />-->
     <PackageReference Include="System.Net.Http" Version="4.3.4" />
   </ItemGroup>
   <ItemGroup>
     <ProjectReference Include="..\OpenSearch.Stack.ArtifactsApi\OpenSearch.Stack.ArtifactsApi.csproj" />
   </ItemGroup>
-</Project>
\ No newline at end of file
+</Project>
diff --git a/opensearch-net-abstractions/src/OpenSearch.OpenSearch.Xunit/OpenSearch.OpenSearch.Xunit.csproj b/opensearch-net/abstractions/src/OpenSearch.OpenSearch.Xunit/OpenSearch.OpenSearch.Xunit.csproj
old mode 100644
new mode 100755
index 3892ea3..77b3d37
--- a/opensearch-net-abstractions/src/OpenSearch.OpenSearch.Xunit/OpenSearch.OpenSearch.Xunit.csproj
+++ b/opensearch-net/abstractions/src/OpenSearch.OpenSearch.Xunit/OpenSearch.OpenSearch.Xunit.csproj
@@ -2,9 +2,9 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
-    <IsPackable>True</IsPackable>
     <Description>Provides an Xunit test framework allowing you to run integration tests against local ephemeral OpenSearch clusters</Description>
     <PackageTags>opensearch,opensearch,xunit,cluster,integration,test,ephemeral</PackageTags>
+    <IsPackable>false</IsPackable>
   </PropertyGroup>
   <ItemGroup>
     <PackageReference Include="xunit" Version="2.4.2" />
@@ -12,4 +12,4 @@
   <ItemGroup>
     <ProjectReference Include="..\OpenSearch.OpenSearch.Ephemeral\OpenSearch.OpenSearch.Ephemeral.csproj" />
   </ItemGroup>
-</Project>
\ No newline at end of file
+</Project>
diff --git a/opensearch-net-abstractions/src/OpenSearch.Stack.ArtifactsApi/OpenSearch.Stack.ArtifactsApi.csproj b/opensearch-net/abstractions/src/OpenSearch.Stack.ArtifactsApi/OpenSearch.Stack.ArtifactsApi.csproj
old mode 100644
new mode 100755
index c228e37..dae96f5
--- a/opensearch-net-abstractions/src/OpenSearch.Stack.ArtifactsApi/OpenSearch.Stack.ArtifactsApi.csproj
+++ b/opensearch-net/abstractions/src/OpenSearch.Stack.ArtifactsApi/OpenSearch.Stack.ArtifactsApi.csproj
@@ -3,6 +3,7 @@
         <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
         <Description>Provides a set of classes to resolve the location of OpenSearch stack products in various stages: released, snapshot and build candidates</Description>
         <PackageTags>opensearch,opensearch,stack,versioning,artifacts</PackageTags>
+        <IsPackable>false</IsPackable>
     </PropertyGroup>

     <ItemGroup>
diff --git a/opensearch-net-abstractions/src/OpenSearch.Stack.ArtifactsApi/Platform/OsMonikers.cs b/opensearch-net/abstractions/src/OpenSearch.Stack.ArtifactsApi/Platform/OsMonikers.cs
old mode 100644
new mode 100755
index 2bf89b5..127b14d
--- a/opensearch-net-abstractions/src/OpenSearch.Stack.ArtifactsApi/Platform/OsMonikers.cs
+++ b/opensearch-net/abstractions/src/OpenSearch.Stack.ArtifactsApi/Platform/OsMonikers.cs
@@ -50,7 +50,7 @@ namespace OpenSearch.Stack.ArtifactsApi.Platform
                        if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX)) return OSPlatform.OSX;
                        if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux)) return OSPlatform.Linux;
                        throw new Exception(
-                               $"{RuntimeInformation.OSDescription} is currently not supported please open an issue @opensearch-project/opensearch-net-abstractions");
+                               $"{RuntimeInformation.OSDescription} is currently not supported please open an issue @opensearch-project/opensearch-net");
                }

                /// <summary>
@@ -78,7 +78,7 @@ namespace OpenSearch.Stack.ArtifactsApi.Platform
                                if (architecture == Architecture.Arm64) return "linux-arm64";
                        }
                        throw new Exception(
-                               $"{RuntimeInformation.OSDescription} is currently not supported please open an issue @opensearch-project/opensearch-net-abstractions");
+                               $"{RuntimeInformation.OSDescription} is currently not supported please open an issue @opensearch-project/opensearch-net");
                }

                public static string GetPlatformArchiveExtension(OSPlatform platform)
@@ -87,7 +87,7 @@ namespace OpenSearch.Stack.ArtifactsApi.Platform
                        if (platform == OSPlatform.OSX) return "tar.gz";
                        if (platform == OSPlatform.Windows) return "zip";
                        throw new Exception(
-                               $"{RuntimeInformation.OSDescription} is currently not supported please open an issue @opensearch-project/opensearch-net-abstractions");
+                               $"{RuntimeInformation.OSDescription} is currently not supported please open an issue @opensearch-project/opensearch-net");

                }
                public static string CurrentPlatformArchiveExtension()
@@ -104,7 +104,7 @@ namespace OpenSearch.Stack.ArtifactsApi.Platform
                        if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux)) return $"{Linux}-{intelX86Suffix}";

                        throw new Exception(
-                               $"{RuntimeInformation.OSDescription} is currently not supported please open an issue @opensearch-project/opensearch-net-abstractions");
+                               $"{RuntimeInformation.OSDescription} is currently not supported please open an issue @opensearch-project/opensearch-net");
                }

                internal static string CurrentPlatformSearchFilter()
@@ -114,7 +114,7 @@ namespace OpenSearch.Stack.ArtifactsApi.Platform
                        if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux)) return "tar";

                        throw new Exception(
-                               $"{RuntimeInformation.OSDescription} is currently not supported please open an issue @opensearch-project/opensearch-net-abstractions");
+                               $"{RuntimeInformation.OSDescription} is currently not supported please open an issue @opensearch-project/opensearch-net");
                }
        }
 }
```

**Comment by wbeckler** [comment]
Process-wise, this looks like a major improvement.

**Review by MaxKsyunz — APPROVED** [review]

**Review by forestmvey — APPROVED** [review]

**Comment by Yury-Fridlyand** [comment]
This PR review confirmed that this PoC has the right approach. All changes were implemented in scope of https://github.com/opensearch-project/opensearch-net/pull/92 and https://github.com/Bit-Quill/opensearch-net/pull/41.

**Pull request closed without merging**